libraries/gcp_backend.rb

```diff
@@ -178,3 +186,203 @@ def camel_case(data)
+  end
+
+  def return_if_object(response)
+    raise "Bad response: #{response.body}" \
+      if response.is_a?(Net::HTTPBadRequest)
+    raise "Bad response: #{response}" \
+      unless response.is_a?(Net::HTTPResponse)
+    return if response.is_a?(Net::HTTPNotFound)
+    return if response.is_a?(Net::HTTPNoContent)
+    result = JSON.parse(response.body)
+    raise_if_errors result, %w[error errors], 'message'
+    raise "Bad response: #{response}" unless response.is_a?(Net::HTTPOK)
+    result
+  end
+
+  def raise_if_errors(response, err_path, msg_field)
+    errors = self.class.navigate(response, err_path)
+    raise_error(errors, msg_field) unless errors.nil?
+  end
+
+  def raise_error(errors, msg_field)
+    raise IOError, ['Operation failed:',
+                    errors.map { |e| e[msg_field] }.join(', ')].join(' ')
+  end
+
+  def build_uri(base_url, template, var_data)
+    URI.join(
+      base_url,
+      expand_variables(template, var_data)
+    )
+  end
+
+  # Allows fetching objects within a tree path.
+  def self.navigate(source, path, default = nil)
+    key = path.take(1)[0]
+    path = path.drop(1)
+    return default unless source.key?(key)
+    result = source.fetch(key)
+    return navigate(result, path, default) unless path.empty?
+    return result if path.empty?
+  end
+
+  def extract_variables(template)
+    template.scan(/{{[^}]*}}/).map { |v| v.gsub(/{{([^}]*)}}/, '\1') }
+            .map(&:to_sym)
+  end
+
+  def expand_variables(template, var_data)
+    extract_variables(template).each do |v|
+      unless var_data.key?(v)
+        raise "Missing variable :#{v} in #{var_data} on #{caller.join("\n")}}"
+      end
+      template.gsub!(/{{#{v}}}/, CGI.escape(var_data[v].to_s))
+    end
+    template
+  end
+end
+
+module Network
+    # A handler for authenticated network request
+  class Base
+    def initialize(link, cred)
+      @link = link
+      @cred = cred
+    end
+
+    def builder
+      Net::HTTP.const_get('Get')
+    end
+
+    def send
+      request = @cred.authorize(builder.new(@link))
+      request['User-Agent'] = generate_user_agent
+      response = transport(request).request(request)
+      unless ENV['GOOGLE_HTTP_VERBOSE'].nil?
+        puts ["network(#{request}: [#{response.code}]",
+              response.body.split("\n").map(&:strip).join(' ')].join(' ')
+      end
+      response
+    end
+
+    def transport(request)
+      uri = request.uri
+      puts "network(#{request}: #{uri})" \
+        unless ENV['GOOGLE_HTTP_VERBOSE'].nil?
+      transport = Net::HTTP.new(uri.host, uri.port)
+      transport.use_ssl = uri.is_a?(URI::HTTPS)
+      transport.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      transport.set_debug_output $stderr \
+        unless ENV['GOOGLE_HTTP_DEBUG'].nil?
+      transport
+    end
+
+    private
+
+    def generate_user_agent
+      "inspec-google/1.0.0"
+    end
+  end
+  class Authorization
+  # A class to aquire credentials and authorize Google API calls.
```

Comment:
Can you put this above line 330? (That seems to be the canonical way of putting a comment describing a class)